cmd/cnbBuild.go

```diff
@@ -565,7 +581,14 @@ func runCnbBuild(config *cnbBuildOptions, telemetry *buildpacks.Telemetry, image
 	}
 
 	creatorArgs = append(creatorArgs, fmt.Sprintf("%s:%s", containerImage, targetImage.ContainerImageTag))
-	err = utils.RunExecutable(creatorPath, creatorArgs...)
+	attr := &syscall.SysProcAttr{
+		Credential: &syscall.Credential{
+			Uid: uint32(uid),
```

Comment:
not supported by windows. Makes the build fail.

Reply:
We never actively supported Windows for cnbBuild anyway. Is anyone running Piper based CI on Windows?

Reply:
That's right regarding the Jenkins pipelines. Nevertheless, we lose the potential to test locally on windows systems which is an option since the beginning of Piper. There are still a bunch of windows boxes out there.

Reply:
@pbusko @c0d1ngm0nk3y @c0d1ngm0nk3y Could we detect the "Running on Windows" case and just do the UID change on Linux?

Reply:
> That's right regarding the Jenkins pipelines. Nevertheless, we lose the potential to test locally on windows systems which is an option since the beginning of Piper. There is still a bunch of windows boxes out there.

@rodibrin  The piper itself is almost with a 100% certainty not working on Windows, and having a special cases in the code (under different build tags) for running unit tests on Windows feels a bit excessive. Wouldn't it make more sense to run unit tests in Docker container to be closer to the actual environment where the binary will be executed?

Reply:
from my POV non-Windows support is not a pre-requisite. As indicated before there are steps which cannot run on Windows today and this has been accepted so far.

Reply:
this prevents to run tests  locally on windows. most of the piper commands run under windows which is still a widely spreaded platform. windows user would need to use docker containers or vms which is cumbersome.

Reply:
FWIW Piper steps are written to expect to be run within a container. I would even consider this potentially dangerous to run them directly on your host system.

Also FWIW I think there is actually a mock for this, isn't there?

Reply:
> windows user would need to use docker containers or vms which is cumbersome.

Is it though? It could be a simple script?

> i would appreciate it if there is a mock or whatever which makes piper on windows buildable.

If that is really desired, I would really love to see some pre-commit check. TBH, it feels strange add technical requirement on some user preferences on local machines. I it is a valid requirement (with I doubt), it should be checked in CI/CD.